### ISSUE TITLE
[no-unexpected-multiline]: False positive with template string

### ISSUE THREAD

**Issue by jacobrask** [typescript-eslint/typescript-eslint#139]
**Repro**

```TS
const IssueTitle = styled.h3<{
  size: StartpageComponentCarousel['params']['size'];
}>`
  display: inline-block;
`;
```

Gives `no-unexpected-multiline` error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `1.1.0` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.12.1` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
The problem here is that the code with a type paremeter produces the following AST:

```TS
styled.h3<{
  size: StartpageComponentCarousel['params']['size'];
}>`
  display: inline-block;
`
```
```CJSON
{
  type: "TaggedTemplateExpression",
  tag: {
    type: "MemberExpression",
    // ...
    loc: {
      start: { line: 1, column: 19 },
      end: { line: 1, column: 28 }
    }
  },
  typeParameter: {
    type: "TSTypeParameterInstantiation",
    // ...
    loc: {
      start: { line: 1, column: 28 },
      end: { line: 3, column: 2 }
    }
  },
  qasi: {
    type: "TemplateLiteral",
    // ...
    loc: {
      start: { line: 3, column: 2 },
      end: { line: 5, column: 1 }
    }
  },
}
```

Notice that the `tag` ends, and the `qasi` starts on different lines. 
Without the type parameter, the two instead end and start on the same line, respectively.

[`no-unexpected-multiline` is hardcoded to ensure this relationship holds true.](https://github.com/eslint/eslint/blob/master/lib/rules/no-unexpected-multiline.js#L73-L78)
We'll have to bring this rule into the plugin in order to fix this, as it's a TS specfic case.

**Comment by bradzacher** [comment]
potentially will be fixed upstream?
https://github.com/eslint/eslint/issues/11650

**Comment by bradzacher** [comment]
This will be fixed by
https://github.com/eslint/eslint/pull/11698

which is approved and will be merged imminently.
I'm going to close this. Feel free to track the final progress over at the PR
